src/api/python/cvc5.pxi

```diff
@@ -186,9 +186,22 @@ cdef class SymbolManager:
     cdef c_SymbolManager* csm
     cdef TermManager tm
 
-    def __cinit__(self, TermManager tm):
-        self.csm = new c_SymbolManager(dereference(tm.ctm))
-        self.tm = tm
+    def __cinit__(self, tm):
+        """
+            Constructor.
+            Initialize with associated Solver or TermManager instance.
+            .. warning:: Initializing with associated solver instance is
+                         deprecated and will be removed in a future release.
+        """
+        if isinstance(tm, TermManager):
+            self.csm = new c_SymbolManager(dereference((<TermManager?>tm).ctm))
+            self.tm = tm
+        # backwards compatibility, deprecated
+        elif isinstance(tm, Solver):
+            self.csm = new c_SymbolManager(dereference((<Solver?>tm).tm.ctm))
+            self.tm = tm.tm
+        else:
+          raise ValueError("Expecting a non-negative integer or string")
```

Comment:
The error message doesn't match. The constructor expects a Solver or a TermManager.